Sources/Common/DataModel/Polygon/Constants.js

```diff
@@ -6,6 +6,13 @@ export const PolygonWithPointIntersectionState = {
   FAILURE: -1,
   OUTSIDE: 0,
   INSIDE: 1,
-  INTERSECTION: 2,
-  ON_LINE: 3,
 };
+export const VTK_DBL_EPSILON = 2.2204460492503131e-16;
+
+export const PolygonWithPolygonIntersectionState = {
+  NO_INTERSECTION: 0,
+  POINT_INTERSECTION: 1,
+  LINE_INTERSECTION: 2,
+};
```

Comment:
why can't you have failure in this case ?
what about the case where the whole polygon is within another polygon

Reply:
One polygon coincident with the other is now implemented.

Reply:
The failure case is when the normal is the zero vector, hence all points are on one line. The computation done would still work on that case since it loops over all edges of one polygon, see if it intersects with the other polygon and does the same on the second polygon if needed.
I would say whether we consider a failure possible depends on whether we want to consider all points being on one line acceptable.
For `pointInPolygon`, the normal being zero vector causes a failure since one cannot computed the winding number if all points are on one line.